// lib-login/esm2015/services/api.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
export class ApiService {
    constructor(http) {
        this.http = http;
    }
    // GET request
    fetchGetUrl(url, data) {
        return this.http.get(url, data);
    }
    // POST request
    fetchPostUrl(url, data = null) {
        return this.http.post(url, data);
    }
    // PUT request
    fetchPutUrl(url, data = null) {
        return this.http.put(url, data);
    }
    // DELETE request
    fetchDelete(url) {
        return this.http.delete(url);
    }
}
ApiService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ApiService, deps: [{ token: i1.HttpClient }], target: i0.ɵɵFactoryTarget.Injectable });
ApiService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ApiService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: ApiService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }]; } });
