// lib-login/esm2015/services/auth.service.js
import { Injectable } from '@angular/core';
import { environment } from '../environments/environment';
import * as i0 from "@angular/core";
import * as i1 from "./api.service";
export class AuthService {
    constructor(apiService) {
        this.apiService = apiService;
        this.environment = environment;
    }
    getCsrfToken() {
        const url = `${this.environment.api.baseUrl}${this.environment.api.auth.login}`;
        return this.apiService.fetchGetUrl(url);
    }
    userAuthentication(data) {
        const url = `${this.environment.api.baseUrl}${this.environment.api.auth.userAuthentication}`;
        return this.apiService.fetchPostUrl(url, data);
    }
    changePassword(data) {
        const url = `${this.environment.api.baseUrl}${this.environment.api.auth.changePassword}`;
        return this.apiService.fetchPostUrl(url, data);
    }
    getJwtToken(data) {
        const url = `${this.environment.api.baseUrl}${this.environment.api.auth.jwtToken}`;
        return this.apiService.fetchPostUrl(url, data);
    }
}
AuthService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AuthService, deps: [{ token: i1.ApiService }], target: i0.ɵɵFactoryTarget.Injectable });
AuthService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AuthService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: AuthService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.ApiService }]; } });
