// lib-login/esm2015/services/user.service.js
import { Injectable } from '@angular/core';
import { environment } from '../environments/environment';
import * as i0 from "@angular/core";
import * as i1 from "./api.service";
export class UserService {
    constructor(apiService) {
        this.apiService = apiService;
        this.environment = environment;
    }
    getUserData() {
        const url = `${this.environment.api.baseUrl}${this.environment.api.auth.getUsers}`;
        return this.apiService.fetchGetUrl(url);
    }
}
UserService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserService, deps: [{ token: i1.ApiService }], target: i0.ɵɵFactoryTarget.Injectable });
UserService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.ApiService }]; } });
