// lib-login/esm2015/state/user/user.store.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { Store, StoreConfig } from '@datorama/akita';
import * as i0 from "@angular/core";
export function createInitialState() {
    return {
        user: {},
        isLoaded: false,
        csrfToken: '',
        jwtToken: ''
    };
}
let UserStore = class UserStore extends Store {
    constructor() {
        super(createInitialState());
    }
};
UserStore.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserStore, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
UserStore.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserStore, providedIn: 'root' });
UserStore = __decorate([
    StoreConfig({
        name: 'user',
        resettable: true
    })
], UserStore);
export { UserStore };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserStore, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return []; } });
