// lib-login/esm2015/state/user/user.query.js
import { Injectable } from '@angular/core';
import { Query } from '@datorama/akita';
import * as i0 from "@angular/core";
import * as i1 from "./user.store";
export class UserQuery extends Query {
    constructor(userStore) {
        super(userStore);
        this.userStore = userStore;
    }
    // get user
    getUserDetail() {
        return this.select(state => state.user);
    }
    // get user data loading status
    getUserLoadingStatus() {
        return this.select(state => state.isLoaded);
    }
    // get CSRF token
    getCsrfToken() {
        return this.select(state => state.csrfToken);
    }
    // get JWT token
    getJwtToken() {
        return this.select(state => state.jwtToken);
    }
    // update user loading status
    updateUserLoadingStatus(status) {
        return this.userStore.update(state => {
            return {
                isLoaded: status
            };
        });
    }
    // update user data
    updateUserData(user = {}) {
        this.userStore.update(state => {
            return {
                user: user,
                isLoaded: true
            };
        });
    }
    // update CSRF token
    updateCsrfToken(token) {
        this.userStore.update(state => {
            return {
                csrfToken: token
            };
        });
    }
    // update JWT token
    updateJwtToken(token) {
        console.log('updupdateJwtToken token ', token);
        return this.userStore.update(state => {
            return {
                jwtToken: token
            };
        });
    }
}
UserQuery.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserQuery, deps: [{ token: i1.UserStore }], target: i0.ɵɵFactoryTarget.Injectable });
UserQuery.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserQuery, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: UserQuery, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: i1.UserStore }]; } });
