// lib-login/esm2015/common/common-material/common-material.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatMenuModule } from '@angular/material/menu';
import { MatTabsModule } from '@angular/material/tabs';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatListModule } from '@angular/material/list';
import { MatGridListModule } from '@angular/material/grid-list';
import * as i0 from "@angular/core";
export class CommonMaterialModule {
}
CommonMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CommonMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CommonMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CommonMaterialModule, imports: [CommonModule,
        MatCheckboxModule,
        MatSidenavModule,
        MatMenuModule,
        MatTabsModule,
        MatProgressSpinnerModule,
        MatListModule,
        MatGridListModule], exports: [MatCheckboxModule,
        MatSidenavModule,
        MatMenuModule,
        MatTabsModule,
        MatProgressSpinnerModule,
        MatListModule,
        MatGridListModule] });
CommonMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CommonMaterialModule, imports: [[
            CommonModule,
            MatCheckboxModule,
            MatSidenavModule,
            MatMenuModule,
            MatTabsModule,
            MatProgressSpinnerModule,
            MatListModule,
            MatGridListModule
        ], MatCheckboxModule,
        MatSidenavModule,
        MatMenuModule,
        MatTabsModule,
        MatProgressSpinnerModule,
        MatListModule,
        MatGridListModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CommonMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    imports: [
                        CommonModule,
                        MatCheckboxModule,
                        MatSidenavModule,
                        MatMenuModule,
                        MatTabsModule,
                        MatProgressSpinnerModule,
                        MatListModule,
                        MatGridListModule
                    ],
                    exports: [
                        MatCheckboxModule,
                        MatSidenavModule,
                        MatMenuModule,
                        MatTabsModule,
                        MatProgressSpinnerModule,
                        MatListModule,
                        MatGridListModule
                    ]
                }]
        }] });
