// lib-login/esm2015/lib/counter/counter.component.js
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
export class CounterComponent {
    constructor() {
        this.counter = 0;
    }
    ngOnInit() {
    }
    onPlusClick() {
        this.counter++;
    }
    onMinusClick() {
        this.counter--;
    }
}
CounterComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CounterComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CounterComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.15", type: CounterComponent, selector: "lib-counter", ngImport: i0, template: "<p>counter works!</p>\n<button (click)=\"onPlusClick()\">+</button>&nbsp;\n<label>{{counter}}</label>&nbsp;\n<button (click)=\"onMinusClick()\">-</button>\n", styles: [""] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: CounterComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'lib-counter',
                    templateUrl: './counter.component.html',
                    styleUrls: ['./counter.component.css']
                }]
        }], ctorParameters: function () { return []; } });
