// lib-login/esm2015/environments/environment.js
// This file can be replaced during build by using the `fileReplacements` array.
// `ng build` replaces `environment.ts` with `environment.prod.ts`.
// The list of file replacements can be found in `angular.json`.
export const environment = {
    production: false,
    api: {
        // baseUrl: "http://localhost:5001",
        baseUrl: "https://solution-finder-demo.capgemini.com:5001",
        auth: {
            login: "/login",
            userAuthentication: "/UserAuthentication",
            changePassword: "/api/changePassword",
            getUsers: "/api/users",
            jwtToken: "/token",
            user_like: "/api/UserFeedBack",
            user_snowlike: "/api/SNOWFeedBack/",
            searchSolution: "/api/SearchSolution",
            searchSolutionDetail: "/api/SearchSolutionById",
            updateUser: "/api/updateUser",
            activeUser: "/api/activateUser/",
            deleteUser: "/api/deleteUser/",
            getUser: "/api/getUser/"
        }
    },
    keys: {
        ENCRYPT_DECRYPT_KEY: "*$0lut!0nf!nder*",
        IV: 'I8zyA4lVhMCaJ5Kg'
    }
};
/*
 * For easier debugging in development mode, you can import the following file
 * to ignore zone related error stack frames such as `zone.run`, `zoneDelegate.invokeTask`.
 *
 * This import should be commented out in production mode because it will have a negative impact
 * on performance if an error is thrown.
 */
// import 'zone.js/plugins/zone-error';  // Included with Angular CLI.
