// lib-login/esm2015/lib/lib-login.module.js
import { NgModule } from '@angular/core';
import { LibLoginComponent } from './lib-login.component';
import { ReactiveFormsModule } from '@angular/forms';
import { NG_ENTITY_SERVICE_CONFIG } from '@datorama/akita-ng-entity-service';
import { AkitaNgDevtools } from '@datorama/akita-ngdevtools';
import { AkitaNgRouterStoreModule } from '@datorama/akita-ng-router-store';
import { HttpClientModule } from '@angular/common/http';
//import { AuthInterceptor } from './interceptor/auth.interceptor';
import { CookieService } from 'ngx-cookie-service';
import { CookieModule } from 'ngx-cookie';
import { CommonMaterialModule } from '../common/common-material/common-material.module';
import { CounterComponent } from './counter/counter.component';
import { LoginComponent } from './login/login.component';
import { environment } from '../environments/environment';
import { UserlistComponent } from './usermanagement/userlist/userlist.component';
import * as i0 from "@angular/core";
import * as i1 from "@datorama/akita-ngdevtools";
import * as i2 from "ngx-cookie";
export class LibLoginModule {
}
LibLoginModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: LibLoginModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LibLoginModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: LibLoginModule, declarations: [LibLoginComponent,
        LoginComponent,
        CounterComponent,
        UserlistComponent], imports: [ReactiveFormsModule,
        CommonMaterialModule, i1.AkitaNgDevtools, i2.CookieModule, AkitaNgRouterStoreModule,
        HttpClientModule], exports: [LibLoginComponent,
        LoginComponent,
        CounterComponent,
        UserlistComponent] });
LibLoginModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: LibLoginModule, providers: [
        { provide: NG_ENTITY_SERVICE_CONFIG, useValue: { baseUrl: 'https://jsonplaceholder.typicode.com' } },
        //{ provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
        CookieService,
    ], imports: [[
            ReactiveFormsModule,
            CommonMaterialModule,
            environment.production ? [] : AkitaNgDevtools.forRoot(),
            CookieModule.forRoot(),
            AkitaNgRouterStoreModule,
            HttpClientModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.15", ngImport: i0, type: LibLoginModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        LibLoginComponent,
                        LoginComponent,
                        CounterComponent,
                        UserlistComponent,
                    ],
                    imports: [
                        ReactiveFormsModule,
                        CommonMaterialModule,
                        environment.production ? [] : AkitaNgDevtools.forRoot(),
                        CookieModule.forRoot(),
                        AkitaNgRouterStoreModule,
                        HttpClientModule
                    ],
                    providers: [
                        { provide: NG_ENTITY_SERVICE_CONFIG, useValue: { baseUrl: 'https://jsonplaceholder.typicode.com' } },
                        //{ provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
                        CookieService,
                    ],
                    exports: [
                        LibLoginComponent,
                        LoginComponent,
                        CounterComponent,
                        UserlistComponent,
                    ]
                }]
        }] });
